File: src/main/java/net/gtaun/shoebill/streamer/data/DynamicCircle.java
package net.gtaun.shoebill.streamer.data;

import net.gtaun.shoebill.data.Vector2D;
import net.gtaun.shoebill.streamer.Functions;

/**
 * Created by Valeriy on 18/4/2016.
 */
public class DynamicCircle extends DynamicArea {

    public DynamicCircle(int id, int playerId) {
        super(id, playerId);
    }

    public static DynamicCircle create(Vector2D location, float size) {
        return create(location, size, -1, -1, -1);
    }

    public static DynamicCircle create(Vector2D location, float size, int worldId, int interiorId, int playerId) {
        DynamicCircle circle = Functions.createDynamicCircle(location, size, worldId, interiorId, playerId);
        areas.add(circle);
        return circle;
    }
}

File: src/main/java/net/gtaun/shoebill/streamer/data/DynamicCircle.kt
package net.gtaun.shoebill.streamer.data

import net.gtaun.shoebill.`object`.Player
import net.gtaun.shoebill.data.Location
import net.gtaun.shoebill.streamer.AllOpen
import net.gtaun.shoebill.streamer.Functions

/**
 * Created by Valeriy on 18/4/2016 in project streamer-wrapper.
 * Copyright (c) 2016 Valeriy. All rights reserved.
 */
@AllOpen
class DynamicCircle(id: Int, player: Player?) : DynamicArea(id, player) {

    companion object {
        @JvmOverloads
        @JvmStatic
        fun create(location: Location, size: Float, player: Player? = null, priority: Int = 0): DynamicCircle {
            val playerId = player?.id ?: -1

            val circle = Functions.createDynamicCircle(location, size, playerId, priority)
            areas.add(circle)
            return circle
        }
    }
}
